app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -12,7 +12,7 @@ module HostManagedExtensions
         alias_method_chain :smart_proxy_ids, :katello
 
         has_one :content_host, :class_name => "Katello::System", :foreign_key => :host_id,
-                               :dependent => :restrict, :inverse_of => :foreman_host
+                               :dependent => :restrict_with_error, :inverse_of => :foreman_host
```

Comment:
if ```ActiveRecord::DeleteRestrictionError``` is an error, then I would think we want to continue to raise an error?

I am seeing in [AR docs](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html)

 >:restrict_with_exception causes an exception to be raised if
>  there are any associated records.

  > :restrict_with_error causes an error to be added to the owner if
  > there are any associated objects.